### DISCUSSION TITLE
Does Bruno support Request polling?

### DISCUSSION THREAD

**Discussion by satya-kaki1** [usebruno/bruno#3196] [Q&A]
I have a request to check a job status. It may takes minutes to finish the job.  How do I keep polling the request until the job is finished?
I have a GET API  {{url}}/api/jobs?jobId=123  which has a Json response.  
           status": { "status": "QUEUED/DONE",},
        
       The Status will be Queued for a min and then it will be done.  Any Suggestions how to handle such scenarios?

**Comment by That-1-QA** [comment]
I've been able to accomplish this with something like the following, make sure to update it with the request name. Maybe not elegant, but it works.

```
const maxNumberOfTries = 10;
const sleepBetweenTries = 5000;

if (!bru.getVar("tries")) {
  console.log("setting tries to 1")
    bru.setVar("tries", 1)
}

test("Poll until complete", function(){
  expect(res.getStatus()).equal(200);
  expect(res.body).haveOwnProperty("status");
  expect(res.body.status).to.be.oneOf(["Pending", "Complete", "Active"]);
})

if ((res.body.status !== "Complete") && (bru.getVar("tries") < maxNumberOfTries)) {
  const tries = parseInt(bru.getVar("tries"), 10);
  bru.setVar("tries", tries + 1);
  console.log(`tries is now ${tries}`)
  await new Promise(r => setTimeout(r, sleepBetweenTries));
  bru.setNextRequest("REPLACE THIS WITH THE REQUEST NAME!");
  } else {
  bru.deleteVar("tries");
}
```
